src/nolongerevil/integrations/mqtt/mqtt_integration.py

```diff
@@ -503,30 +510,50 @@ async def _publish_ha_state(
                 retain=True,
             )
 
-        # Target temperature
-        if shared_values.get("target_temperature") is not None:
+        # Target temperatures - publish based on mode
+        # In heat_cool mode: publish high/low but NOT single target_temperature
+        # In heat/cool/off mode: publish single target_temperature but NOT high/low
+        if ha_mode == "heat_cool":
```

Comment:
Seeing the same  `if` here -> code smell - maybe we should have a different architecture.  a `consts` file that has the Modes as a named Enum type; then a list of topics that ought to be published given a `mode` ?

Reply:
yep, instead of "if mode is x" I would like a "for allowed_topic in mode" and we must always pass in a constant array of allowed_topics defined in a const